projects/plugins/wpcomsh/custom-colors/colors.php

```diff
@@ -1483,17 +1483,33 @@ public static function print_theme_css() {
 	}
 
 	/**
-	 * Print block editor CSS.
+	 * Adds block editor CSS to the block editor settings.
+	 *
+	 * @param array $editor_settings Block editor settings.
 	 */
-	public static function print_block_editor_css() {
+	public static function add_block_editor_css( $editor_settings ) {
 		if ( ! self::should_enable_colors() ) {
-			return;
+			return $editor_settings;
 		}
+
 		$css = self::get_theme_css();
 
-		wp_register_style( 'custom-colors-editor-css', false, array(), '20210311' ); // Register an empty stylesheet to append custom CSS to.
-		wp_enqueue_style( 'custom-colors-editor-css' );
-		wp_add_inline_style( 'custom-colors-editor-css', $css ); // Append inline style to our new stylesheet
+		if ( is_array( $editor_settings['styles'] ) ) {
```

Comment:
I think the code here can be simplified by initializing the array to avoid duplication:

 ```
if ( is_array( $editor_settings['styles'] ) ) {
   $editor_settings['styles'] = array();
}
```